gcloud-java-core/src/main/java/com/google/gcloud/AuthCredentials.java

```diff
@@ -42,8 +38,66 @@
   private static class AppEngineAuthCredentials extends AuthCredentials {
 
     private static final AuthCredentials INSTANCE = new AppEngineAuthCredentials();
```

Comment:
We can't have the INSTANCE anymore as we also maintain a state (`scopesRequired` and `scopes`).
What you need to do is to change `capture()` to return an instance of `AppEngineAuthCredentialsState`
that store both variables and use them to create a new `AppEngineAuthCredentials` instance in `restore()`.